sdks/java/core/src/main/java/org/apache/beam/sdk/coders/RowCoderGenerator.java

```diff
@@ -285,15 +289,24 @@ public InstrumentedType prepare(InstrumentedType instrumentedType) {
     // per-field Coders.
     static Row decodeDelegate(Schema schema, Coder[] coders, InputStream inputStream)
         throws IOException {
+      int fieldCount = VAR_INT_CODER.decode(inputStream);
```

Comment:
Can we leave this out by just using the length of the BitSet?

Reply:
Unfortunately no. BitSet.size() returns the total space used, which rounds up to the next word size.

Reply:
as discussed, we don't want to have to reorder fields in the encoding.